api/applications/views.py

```diff
@@ -0,0 +1,89 @@
+"""
+Views related to OAuth2 platform applications. Intended for OSF internal use only
+"""
+from rest_framework.exceptions import APIException
+from rest_framework import generics
+from rest_framework import renderers
+from rest_framework import permissions as drf_permissions
+
+from modularodm import Q
+
+from framework.auth import cas
+from website.models import ApiOAuth2Application
+
+from api.base.filters import ODMFilterMixin
+from api.base.utils import get_object_or_error
+from api.applications.permissions import OwnerOnly
+from api.applications.serializers import ApiOAuth2ApplicationSerializer
+
+
+class ApplicationList(generics.ListCreateAPIView, ODMFilterMixin):
+    """
+    Get a list of API applications (eg OAuth2) that the user has registered
+    """
+    permission_classes = (
+        drf_permissions.IsAuthenticated,
+    )
+
+    serializer_class = ApiOAuth2ApplicationSerializer
+
+    renderer_classes = [renderers.JSONRenderer]  # Hide from web-browsable API tool
```

Comment:
This endpoint includes sensitive information (client secrets) that should not be exposed to users outside the OSF

In a separate followon PR, the plan is to remove this from swagger, restrict to the admin scope, and generally discourage external visibility.